src/main/kotlin/app/revanced/patches/shared/settings/preference/impl/ArrayResource.kt

```diff
@@ -11,6 +11,7 @@ import org.w3c.dom.Element
  *  @param name The name of the array resource.
  *  @param items The items of the array resource.
  */
+// TODO: allow specifying an array resource file instead of using a list of StringResources
```

Comment:
Ideally we can get rid of exposing StringResource as an API entirely. Conventionlly, the key starts with "revanced" followed by some other string that can be inferred through the current setting (such as the setting key). By applying that convention internally, we can directly use strings instead of StringResource.